js/communication/geppettoJupyter/PythonControlledCapability.js

```diff
@@ -163,13 +178,16 @@ define(function (require) {
             default:
               break;
             }
+            // Don't sync if new value is emtpy string
             if (newValue !== '') {
-              this.syncValueWithPython(newValue);
-              if (this.props.callback) {
-                this.props.callback(newValue, this.oldValue === undefined ? this.state.value : this.oldValue);
-                this.oldValue = newValue;
-              }
+              this.syncValueWithPython(newValue, this.oldValue);
```

Comment:
just curious about this, are we now comparing new and old value in the handler function provided?

Reply:
Also, could we add some comment here on why are we passing new and old so at least who read this code for the first time has an idea of what happens underneath? Thx.

Reply:
My bad. Removed the line. Probably I mixed up with the callback